### ISSUE TITLE
Beginner Issues 

### ISSUE THREAD

**Issue by satu0king** [CircuitVerse/CircuitVerse#2]
These are just few ideas or bugs I know right now. I will keep updating the list. Before starting anything, create an issue and await instructions. Feel free to suggest any features or bug fixes

Simulator
- [ ] Counter 
- [ ] Decimal to BCD and vice versa 
- [ ] Combinational Analysis - Make it easier to give input. 
- [x] 7 Segment - change color 
- [ ] Add undo - redo buttons 
- [ ] Undo - fixes - combinational and new project loaded: call backup()
- [ ] Test and Fix shortcuts in firefox 
- [x] Move from this to prototype 
- [x] Complete Plotting implementation
- [ ] Disable copying / combinational circuits according to assignment preferences 
- [ ] Watermark with link in embed - outside only 
- [ ] Add scroll bar to sidebar 
- [ ] Add a link to assignment desription on the simulator page
- [ ] Enable minimap to show rotated elements
- [ ] Fix Trivial cases - constant and Z state in combinational analysis  
- [ ] Configure shortcuts for Zoom/save online 
- [ ] Setup a good offline backing of circuits

bug fixes 
- [ ] Save empty circuit causes problems 
- [ ] Subcircuit of empty circuit causes problems 
- [ ] Image of empty circuit causes problems 
- [ ] Undo after deleting subcircuit (ie main circuit was deleted) causes problems. 

Ambitious Features 
- [ ] Expresso - replacement of combinational analysis 
- [x] Custom layout for sub circuits - logisim does a brilliant job at this 
- [ ] Write Test suite for simulator

Website - rails + normal front end 
- [ ] Configure back button correctly 
- [x] Flash messages ( notice ) not configured correctly 
- [ ] Assignment Grading system
- [ ] Customize Dashboard 
- [ ] Add shortcuts to getting started page
- [ ] Add a link to view project in assignment page
- [ ] Trending and Featured sections 
- [ ] Preview should go full screen
- [ ] Email for assignment submissions for students. Assignment report for mentor 
- [ ] Use CSS and beautify email
- [ ] Follow users ( probably later )
- [ ] Write Test suite for backend
- [ ] Countries and University - to users
- [ ] Add shortcut on homepage for pending assignments
- [ ] Add shortcut on simulator page for assignment description

Others 
- [ ] Powerpoint Plug-In

Documentation 
This involves a full documentation of all circuit elements. It shouldn't take more than a days work. It is important to someone who comes to our website and has little knowledge of the various components. 
Things to document. 

1. What the element does
2. Truth table if applicable 
3. Small demo 
4. Uses of the element 
5. Further reading resources

**Comment by mohammedsafwat** [comment]
I have fixed a bug related to saving an empty circuit. Kindly review and inform me if everything is correct.

**Comment by tachyons** [comment]
Can we split these issues into separate github issues

**Comment by satu0king** [comment]
> Can we split these issues into separate GitHub issues

Yes, we need to kill this issue and split it across separate issues. Some of these are resolved and need to be marked as completed.

**Comment by utkarshg99** [comment]
@satu0king I can help with the documentation part, can you guide me how to get started after setting up of my development environment?
From what I have understood I think you need a separate webpage with description of all elements on your site. Am I right?

**Comment by satu0king** [comment]
> @satu0king I can help with the documentation part, can you guide me on how to get started after setting up of my development environment?
> From what I have understood I think you need a separate webpage or a pdf with a description of all elements on your site. Am I right?
> I am making the documentation content like this: https://docs.google.com/document/d/1i9JRlZcfrVNDnx7fDwTlFFkjljNuadqTgiHGCAHxt0A/edit?usp=sharing
> Please have a look.

We already have a documentation: docs.circuitverse.org 
See https://github.com/CircuitVerse/CircuitVerseDocs

**Comment by Sadat97** [comment]
I would love to work on the assignment grading system.. am currently working on an LMS system, I have a bit experience in this area. but I need more details on this issue. is there a requirement or a story I can get more info on what to do?

**Comment by felixfaisal** [comment]
Hey it's my first time, I'd like to work on decimal to bcd and vice versa. Please help me out.

**Comment by falguniraina** [comment]
Hi!! this is the first time i'm contributing somewhere i would love to try my hands in documentation please guide me how to get started!!

**Comment by Shivansh2407** [comment]
Hey @satu0king Sir the Redo option is not available in the menu yet. I would like to give a try to it .

**Comment by cognitive137** [comment]
Hello, where is the shortcut to getting started required?

**Comment by Kushagraw12** [comment]
@satu0king  Can i add breadcrumbs to the website that will make navigation easier from one page to another?

**Comment by Roshan13046** [comment]
I would like to contribute to this issue.
Is the issue open?

**Comment by rohan2734** [comment]
@satu0king I would like to work on the task ```Decimal to BCD and vice versa```  please assign it to me,

**Comment by anuragdandge** [comment]
@satu0king  i would like work and contribute on "configure back button "  can i work on it

**Comment by harshitaaaaaa** [comment]
@satu0king I would like to work on the customization of dashboard. Can I go ahead?

**Comment by Just-A-Pixel** [comment]
@satu0king I would like to work and contribute on 'Add shortcuts to getting started page' and/or 'Preview should go full screen' . Can I go ahead?

**Comment by satu0king** [comment]
@all this is an old Issue created 2 years back. I don't know if these are relevant anymore. Check if they are relevant and create a new issue to work on them.

**Comment by surbhigoel191** [comment]
@satu0king Should the decimal-bcd converter be in Tools or as a circuit element?

**Comment by Palaksharma23** [comment]
@satu0king  i would like to work and contribute to 'Add scroll bar to sidebar' . can i go ahead ?

**Comment by UTKARSHSHARMA718** [comment]
Hi sir/mam, I am utkarsh, new to open source, and looking forward to enhancing and learning new skills and gaining some experience. And I would love to contribute to your organization.  Since I had used circuit verse in my 2nd semester for my electrical lab I am a bit familiar with it so, can I work on ''Decimal to BCD and vice versa'' please?

**Comment by SOUMOJIT-CHAKRABORTY** [comment]
Hi! I am new to opensource can anyone suggest from where I should start my journey?

**Comment by SOUMOJIT-CHAKRABORTY** [comment]
@satu0king Sir I am new to opensource I want to work on Decimal to BCD and vice versa. Can you assign me this.

**Comment by RahulSingh212** [comment]
@satu0king  If the issue is still open, I would like to work on it.

**Comment by tachyons** [comment]
If you want to work on any of the above tasks, create a new issue and refer that issue from your PR. Also ensure that issue is not created before to avoid duplicates

**Comment by ZaidKhan2002** [comment]
Hey @satu0king could you please update this list or it's already been updated ?

**Comment by akashroy1** [comment]
Hello! Can I work on this issue or it is already assigned?

**Comment by Palaksharma23** [comment]
> If you want to work on any of the above tasks, create a new issue and refer that issue from your PR. Also ensure that issue is not created before to avoid duplicates

@akashroy1 yes you can go ahead, please follow this instruction

**Comment by akashroy1** [comment]
Thanks for your reply.

**Comment by Ranit-Bandyopadhyay** [comment]
i want to work on this issues. Please assign me this issue

**Comment by Prerna-0202** [comment]
> Hey @satu0king , I would like to work on this issue This would be my first contribution, I want to work on this issue, Please assign This to me.

@Nikuunj   To begin working on any of the mentioned tasks, open a new issue and refer to it with your PR. Also, ensure that the issue is present right now.

**Comment by Stitipragyanbarik** [comment]
@satu0king  Hi! I would like to work on this issue. The problem seems to be with the undo functionality after a main or subcircuit is deleted. I will investigate how the undo feature interacts with these deletions and make the necessary fixes to ensure it works as expected.

I will also run tests to verify that the fix works properly and ensure compatibility across different browsers.

Looking forward to working on this!

**Comment by ChandrakantaMandal** [comment]
Hi maintainers 👋
I’m interested in working on this issue and would love to contribute. If the issue is still open, please assign it to me.

**Comment by cognitive137** [comment]
Hi circuitVerse/CircuitVerse,

Can I please be removed from this list?

Thank you

Kind regards,
Sana
